api/src/main/java/org/openmrs/api/impl/PersonServiceImpl.java

```diff
@@ -182,9 +182,12 @@ public PersonAttributeType savePersonAttributeType(PersonAttributeType type) thr
 		
 		PersonAttributeType attributeType = dao.savePersonAttributeType(type);
 		
-		if (updateExisting) {
-			//we need to update index in case searchable property has changed
-			Context.updateSearchIndexForType(PersonAttribute.class);
+		if (updateExisting ) {
+			Boolean oldSearchable = dao.getSavedPersonAttributeTypeSearchable(type);
+			if (oldSearchable != type.getSearchable()) {
```

Comment:
Not sure about the != comparison @mogoodrich ?  Does this do the right thing with Boolean objects?

Reply:
I believe it worked when I tested, @mseaton , but good point... and it should probably be null-safe as well... will tweak and re-test.

Reply:
Should do the needful. You would only get in trouble with nulls, but the `searchable` field is not nullable in the database.

Reply:
I'll switch to .equals... since searchable is not nullable, I'm inclined to skip the null checks and it just clutters the code.